storage/rocksdb/ib_ut0counter.h

```diff
@@ -84,7 +84,9 @@ struct thread_id_indexer_t : public generic_indexer_t<Type, N> {
   /* @return a random number, currently we use the thread id. Where
   thread id is represented as a pointer, it may not work as
   effectively. */
-  size_t get_rnd_index() const { return get_curr_thread_id(); }
+  size_t get_rnd_index() const {
+    return reinterpret_cast<std::uintptr_t>(get_curr_thread_id());
```

Comment:
@percona-ysorokin the function returns `size_t` but the cast is to `std::uintptr_t`

Reply:
@laurynas-biveinis `get_curr_thread_id()` is basically `pthread_self()` that returns `pthread_t`. On all Linuxes and  MacOSes I tested, `pthread_t` was either `void *` or `<some_opaque_struct> *`, a pointer to something in other words. In the standard `std::uintptr_t` is defined as an unsigned type with the same size as a pointer. That is why it is safe to use `reinterpret_cast<std::uintptr_t>()` here. After that, that `std::uintptr_t` can be easily implicitly converted to `std::size_t` (which is either `unsigned long` or `unsigned long long`).
Moreover, although I did not test this, I believe the same construct should work o Windows as well where `get_curr_thread_id()` is  `GetCurrentThreadId()` that returns `DWORD`.
So, it is intentional.